module/trace/trace.go

```diff
@@ -112,141 +132,131 @@ func (t *OpenTracer) Done() <-chan struct{} {
 	ctx context.Context,
 	collectionID flow.Identifier,
 	spanName SpanName,
-	opts ...opentracing.StartSpanOption) (opentracing.Span, context.Context, bool) {
-
+	opts ...trace.SpanStartOption,
+) (trace.Span, context.Context, bool) {
 	if !collectionID.IsSampled(t.sensitivity) {
-		return &NoopSpan{&NoopTracer{}}, ctx, false
+		return t.noopSpan, t.noopCtx, false
 	}
 
-	rootSpan := t.entityRootSpan(collectionID, EntityTypeCollection)
-	ctx = opentracing.ContextWithSpan(ctx, rootSpan)
+	ctx, rootSpan := t.entityRootSpan(ctx, collectionID, EntityTypeCollection)
 	return t.StartSpanFromParent(rootSpan, spanName, opts...), ctx, true
 }
 
 // StartTransactionSpan starts a span that will be aggregated under the given transaction.
 // All spans for the same transaction will be aggregated under a root span
-func (t *OpenTracer) StartTransactionSpan(
+func (t *Tracer) StartTransactionSpan(
 	ctx context.Context,
 	transactionID flow.Identifier,
 	spanName SpanName,
-	opts ...opentracing.StartSpanOption) (opentracing.Span, context.Context, bool) {
-
+	opts ...trace.SpanStartOption,
+) (trace.Span, context.Context, bool) {
 	if !transactionID.IsSampled(t.sensitivity) {
-		return &NoopSpan{&NoopTracer{}}, ctx, false
+		return t.noopSpan, t.noopCtx, false
 	}
 
-	rootSpan := t.entityRootSpan(transactionID, EntityTypeTransaction)
-	ctx = opentracing.ContextWithSpan(ctx, rootSpan)
+	ctx, rootSpan := t.entityRootSpan(ctx, transactionID, EntityTypeTransaction)
 	return t.StartSpanFromParent(rootSpan, spanName, opts...), ctx, true
 }
 
-func (t *OpenTracer) StartSpanFromContext(
+func (t *Tracer) StartSpanFromContext(
 	ctx context.Context,
 	operationName SpanName,
-	opts ...opentracing.StartSpanOption,
-) (opentracing.Span, context.Context) {
-	parentSpan := opentracing.SpanFromContext(ctx)
-	if parentSpan == nil {
-		return &NoopSpan{&NoopTracer{}}, ctx
-	}
-	if _, ok := parentSpan.(*NoopSpan); ok {
-		return &NoopSpan{&NoopTracer{}}, ctx
-	}
-
-	opts = append(opts, opentracing.ChildOf(parentSpan.Context()))
-	span := t.Tracer.StartSpan(string(operationName), opts...)
-	return span, opentracing.ContextWithSpan(ctx, span)
+	opts ...trace.SpanStartOption,
+) (trace.Span, context.Context) {
+	ctx, span := t.tracer.Start(ctx, string(operationName), opts...)
+	return span, ctx
 }
 
-func (t *OpenTracer) StartSpanFromParent(
-	span opentracing.Span,
+func (t *Tracer) StartSpanFromParent(
+	parentSpan trace.Span,
 	operationName SpanName,
-	opts ...opentracing.StartSpanOption,
-) opentracing.Span {
-	if _, ok := span.(*NoopSpan); ok {
-		return &NoopSpan{&NoopTracer{}}
+	opts ...trace.SpanStartOption,
+) trace.Span {
+	if !parentSpan.SpanContext().IsSampled() {
+		return t.noopSpan
 	}
-	opts = append(opts, opentracing.ChildOf(span.Context()))
-	return t.Tracer.StartSpan(string(operationName), opts...)
+	ctx := trace.ContextWithSpan(context.Background(), parentSpan)
+	_, span := t.tracer.Start(ctx, string(operationName), opts...)
+	return span
```

Comment:
is `ContextWithSpan` automatically label the span relationship as children/parent ?

Reply:
yep!